python/lsst/ts/MTAlignment/alignment_csc.py

```diff
@@ -144,191 +163,356 @@ async def configure(self, config):
-        aligned = False
-        loopcount = 1
-        while not aligned:
-            self.log.info(f"Loop iteration: {loopcount}")
-            await self.model.measure_target("CAM")
-            await self.model.measure_target("M2")
-
-            cam_offset = self.parse_offsets(await self.model.get_target_offset("CAM"))
-            m2_offset = self.parse_offsets(await self.model.get_target_offset("M2"))
-
-            if self.in_tolerance(cam_offset) and self.in_tolerance(m2_offset):
-                break
-            elif loopcount >= self.max_iters:
-                break
-            else:
-                loopcount += 1
 
-    def parse_offsets(self, t2sa_string):
-        """
-        Takes a string containing spatial coordinates  from T2SA, and returns
-        a dict with the following keys: RefFrame, X, Y, Z, Rx, Ry, Rz, and
-        Timestamp
+        if target != "M1M3":
+            await self.model.measure_target(target)
 
-        Parameters
-        ----------
+        target_offset_str = await self.model.get_target_offset(
+            target=target, reference_pointgroup="M1M3"
+        )
+        target_offset = parse_offsets(target_offset_str)
 
-        t2sa_string : `str`
-            the ascii string from T2SA. We expect an ascii string delimited
-            with colons and semicolons, formatted like this:
+        if target_offset is None:
+            raise RuntimeError(
+                f"Failed to parse offsets for {target}: {target_offset_str}"
+            )
 
-            <s>;X:<n>;Y:<n>;Z:<n>;Rx:<n>;Ry:<n>;Rz:<n>;<date>
+        await self.evt_offsetsPublish.set_write(**target_offset)
 
-            where <s> is the name of the reference frame and <n> is a
-            floating point value.
-        """
-        bits = [b.split(":") for b in t2sa_string.split(";")]
-        coordsDict = {}
-        try:
-            # ref frame
-            coordsDict["RefFrame"] = bits[0][1]
-
-            # coords
-            for s in bits[1:7]:
-                coordsDict[s[0]] = float(s[1])
-
-            # timestamp
-            coordsDict["Timestamp"] = f"{bits[7][0]}:{bits[7][1]}:{bits[7][2]}"
-        except ValueError or IndexError:
-            raise Exception(
-                f"Failed to parse coordinates string '{t2sa_string}' received from T2SA."
-            )
-        return coordsDict
+    async def run_telemetry_loop(self) -> None:
+        """Run telemetry loop."""
+
+        assert self.model is not None
 
-    def in_tolerance(self, coords):
+        while self.disabled_or_enabled:
+
+            status = await self.model.check_status()
+            if status == "READY":
+                self.laser_status_ready.set()
+            else:
+                self.laser_status_ready.clear()
+
+            # TODO: Publish events with telemetry data.
+
+            await asyncio.sleep(self.heartbeat_interval)
+
+    async def reset_telemetry_loop(self) -> None:
+        """Reset telemetry loop."""
```

Comment:
It's a good point about just cancelling the telemetry loop. I think it would most likely be ok. Although, I have had problems in the past with socket servers (especially on Windows) handling cancellations.